### ISSUE TITLE
Mouse interation is different between 2D and 3D views

### ISSUE THREAD

**Issue by haesleinhuepf** [napari/napari#2060]
## 🐛 Bug
When using the mouse for changing the view in 2D and 3D, functions are different. It's "complicated" to pan in 3D while there are multiple easy ways for zooming in 2D and 3D. I'm wondering if this is intentional.

## To Reproduce

1. Open any 3D dataset in 2D view
2. zoom and pan (left mouse drag&drop) the dataset
3. Switch to 3D view
4. zoom and pan (SHIFT + left mouse drag&drop) the dataset

## Expected behavior

The mouse wheel is intuitive for zooming. Thus, I suggest changing the function of the right mouse button:
2D, right mouse button: rotate dataset around Z axis (currently it zooms on the right mouse button)
3D, right mouse button: rotate/tilt dataset (currently SHIFT + left mouse button does that)
Then, in 3D view, the left mouse button could be panning (without SHIFT being pressed)

## Environment

I'm using Windows. Experience on Mac may be different depending on how the right mouse button is configured.

## Additional context

IMHO, mouse buttons should not change their functionality when switching between 2D and 3D. Furthrmore, I believe zooming, panning and rotating/tilting should be possible with mouse only and without touching the keyboard.

**Comment by sofroniewn** [comment]
We've just been using the ArcBall Camera from vispy with its defaults in 3D with. Agreed the button mapping may not be so intuitive, I think for a long time I didn't even realize Shift dragging was possible. There are also some other cameras like the ly camera that we want to support #1877. So I'm very open to improving these things, just needs some dedicate thought about what the best options defaults are and so am curious what others think too! cc @royerloic who does a lot with 3D I think :-)

**Comment by jni** [comment]
imho the current left-click behaviour is intuitive in both cases (rather than intuitive: it matches my expectations of what they should do based on other software). The lack of discoverability about 3D panning is a problem, but I don't think harmonising panning between 2D and 3D is desirable. And anyway it sort of already is harmonised, if you think of 3D as "panning" in a spherical world — which matches e.g. what Google Earth does. 

So, I'm -1 on making left click pan in 3D, but +1 on brainstorming about how to make 3D translation easier to discover.

I'm also not sure whether we want rotation about z in 2D to be a first-class citizen...

**Comment by alisterburt** [comment]
I agree with @jni that the left click and drag behaviour is what I would expect

big +1 from me on 3D panning needing to move away from a keyboard + mouse combo, it's been a little pain point for me too (should've raised an issue)

I use [ChimeraX](https://www.cgl.ucsf.edu/chimerax/) quite a lot and watching people use that for the first time I've never seen anyone have any particular trouble

The defaults there are
- left click and drag: move camera around current center of rotation
- right click and drag (default): move camera perpendicular to camera rays
- middle click and drag: move camera perpendicular to camera rays

The right mouse click and drag behviour is configurable (via gui buttons) to a bunch of different things (maybe too many, I only use a few)
![image](https://user-images.githubusercontent.com/7307488/103802243-c4864a00-5046-11eb-9dfe-b7640b10cf94.png)
 
Configuring right mouse click and drag to "move active object" changes the behabiour from movement of the camera to movement of the object within the scene. This makes relative positioning of objects in 3D quite natural, you can rotate the scene with the left mouse button, move an "active object" or group of objects with the right mouse button... small demo below

![Peek 2021-01-06 17-41](https://user-images.githubusercontent.com/7307488/103802692-71f95d80-5047-11eb-8c3d-13ff5ec9d898.gif)

I really like having the panning option still available with middle click and drag whilst the right mouse click and drag is assigned to moving active objects.

I tried to configure napari this way a while ago using https://github.com/napari/napari/blob/master/examples/custom_mouse_functions.py as a starting point but couldn't figure out what to call to have it mirror the shift + left click behaviour

Curious to hear any thoughts!

**Comment by haesleinhuepf** [comment]
> The right mouse click and drag behviour is configurable (via gui buttons)

Maybe we could desgin a small dock widget that shows mouse configuration and allows changing it? I did something like that (a decade ago):
![image](https://user-images.githubusercontent.com/12660498/103820194-2c9b5700-506c-11eb-9059-a72ebc28afc8.png)

**Comment by alisterburt** [comment]
That would definitely be cool!

I think something like the existing `help -> show keybindings` menu but for mouse bindings would probably be useful too? I don't know if having a dock widget for modifying mouse bindings and a separate `show mouse bindings` floating window makes sense. 

Maybe either dock widget or window for both (adding an edit mechanism to the floating window? making the window into a dock widget?)?

It seems like wanting configurability vs. having more discoverable, intuitive defaults in 3d rendering mode are slightly separate issues, with the defualts being probably a quicker fix requiring less thought?

I had a quick look at changing the defaults but had trouble figuring out how all the events connect up. From what I could see, the default behaviour should probably be modified [here](https://github.com/napari/napari/blob/1a0dad4ac1e08f99dcfed645544cdf4bc5d06294/napari/_vispy/vispy_canvas.py) 

Looking at the [vispy canvas](https://github.com/vispy/vispy/blob/master/vispy/scene/canvas.py) I can't quite figure out where the modifier key is taken into account and it looks like mouse wheel click events aren't already received

Do people agree on a middle mouse click and drag = move camera, right mouse click and drag = move camera (but configurable) system?

**Comment by psobolewskiPhD** [comment]
FYI I've made a napari/docs PR that updates the Viewer tutorial to cover the 3D mode mouse interactions: https://github.com/napari/docs/pull/159/files
